### PR TITLE
CSD-316 new function to convert networkx to GraphData object

### PR DESCRIPTION
**Pull request by kibidi**
Added a function that converts a networkx data object to the required GraphData object. Also added a test and one of the dependent functions broke

### CODE REVIEW

File: probinet/input/loader.py [hunk 1]
```diff
@@ -21,6 +22,40 @@
 from ..models.classes import GraphData
 
 
+def build_adjacency_from_networkx(
+    network: nx.Graph,
+    edge_weight: list[str],
+    file_name: Optional[str] = None,
+    # TODO: add csv file name
```
Comment (diegoabt) [inline probinet/input/loader.py:29]: We should remove this to-do. I think it is already implemented.

File: probinet/input/loader.py [hunk 2]
```diff
@@ -21,6 +22,40 @@
 from ..models.classes import GraphData
 
 
+def build_adjacency_from_networkx(
+    network: nx.Graph,
+    edge_weight: list[str],
```
Comment (diegoabt) [inline probinet/input/loader.py:27]: Could you please rename this variable to something like `weight_list`?

File: probinet/input/loader.py [hunk 3]
```diff
@@ -21,6 +22,40 @@
...
+    # TODO: add csv file name
+) -> GraphData:
+    """
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+    """
```
Comment (diegoabt) [inline probinet/input/loader.py:38]: Could you please fill this section out? Any simple description would work. :smile:

File: probinet/input/loader.py [hunk 4]
```diff
@@ -21,6 +22,40 @@
 from ..models.classes import GraphData
 
 
+def build_adjacency_from_networkx(
+    network: nx.Graph,
+    edge_weight: list[str],
+    file_name: Optional[str] = None,
```
Comment (diegoabt) [inline probinet/input/loader.py:28]: Can you use a `Path` object here, instead of a `str`?

File: tests/test_input.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+"""
+
+import unittest
+import networkx as nx
+from probinet.input.loader import build_adjacency_from_networkx
+from probinet.models.classes import GraphData
+
+
+class TestInput(unittest.TestCase):
+
```
Comment (diegoabt) [inline tests/test_input.py:11]: Somewhere here we should add a `setUp` method that creates a temporary folder, and then pass it to the function we are testing. This folder disappears once the test is finished. That way, we won't be left with some auxiliary files every time the test is run.

File: tests/test_input.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+"""
+"""
```
Comment (diegoabt) [inline tests/test_input.py:2]:
```suggestion
"""
Test cases for the input module.
"""
```

File: tests/test_input.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+            G.edges[edge]["weight2"] = i
+
+        g_data = build_adjacency_from_networkx(G, edge_weight=["weight", "weight2"])
+
+        self.assertTrue(type(g_data) is GraphData)
+        # XXX: test didn't work; weights not added to graph using build_adjacency_from_file
+        # for n, w_name in enumerate(["weight","weight2"]):
+        #     for n1,n2,weight in g_data.graph_list[n].edges(data=w_name):
+        #         w = G.edges[(n1,n2)][w_name]
+        #         self.assertEqual(w, weight)
```
Comment (diegoabt) [inline tests/test_input.py:26]: Thank you for this! I will work on this bug after we close this PR, and uncomment this test.

**Review by diegoabt — COMMENTED** [review]
Great work! Thank you for proposing this function; very useful! I left a few suggestions that would be nice to implement before adding this feature into the package. I'll be happy to discuss this further. :smiley:

**Comment by diegoabt** [comment]
@kibidi , you might need to sync your fork before working on the review. We've added a couple of new features into the `dev` branch since you opened the PR.

File: probinet/input/loader.py [hunk 8]
```diff
@@ -11,7 +11,7 @@
 import networkx as nx
```
Comment (jcpassy) [inline probinet/input/loader.py:11]: Imports are not sorted properly, I believe.

File: probinet/input/loader.py [hunk 9]
```diff
@@ -21,6 +21,50 @@
...
+    file_name
+        name of csv file (and path) created from networkx graph (used to create GraphData object)
+        e.g. /path/to/file/file_name.csv
+    Returns
+    -------
+    GraphData
+        GraphData object created from networkx graph
+    """
+    attribute_names = {
+        key for _, _, data in network.edges(data=True) for key in data.keys()
```
Comment (jcpassy) [inline probinet/input/loader.py:47]: `data` is a dict, so this is enough:
```python
for key in data
```

File: probinet/input/loader.py [hunk 10]
```diff
@@ -21,6 +21,50 @@
...
+    }
+    for w in weight_list:
+        assert w in attribute_names, f"{w} is not an attribute"
+
+    if not file_name or Path(file_name).suffix != ".csv":
+        file_name = Path.cwd()/"edge_list.csv"
```
Comment (jcpassy) [inline probinet/input/loader.py:53]: Please add a single space on both sides of the `/` operator (PEP 8):

```python
         file_name = Path.cwd() / "edge_list.csv"
```

File: tests/test_input.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
+"""
+Test cases for the input module.
+"""
+
+import unittest
+import networkx as nx
+from probinet.input.loader import build_adjacency_from_networkx
+from probinet.models.classes import GraphData
+from .fixtures import BaseTest
```
Comment (jcpassy) [inline tests/test_input.py:9]: Imports are not sorted properly.

File: tests/test_input.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    def test_build_adjacency_from_networkx(self):
+        G = nx.erdos_renyi_graph(n=14, p=0.5, seed=7, directed=False)
+        # Add weights
+        for i, edge in enumerate(G.edges()):
+            G.edges[edge]["weight"] = 3
+            G.edges[edge]["weight2"] = i
+
+        g_data = build_adjacency_from_networkx(G, weight_list=["weight", "weight2"],file_name=self.folder+"edges.csv")
+
+        self.assertTrue(type(g_data) is GraphData)
```
Comment (jcpassy) [inline tests/test_input.py:23]: I would probably use `self.assertIsInstance` here, or if you are really worry about the type `self.assertIs` as described in the [documentation](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIsInstance).

**Review by jcpassy — APPROVED** [review]
Thanks a lot @kibidi for your contribution. :)
My comments are only minor ones and related to style and standards. Great job! 👍

File: probinet/input/loader.py (hunk 8 above)
Comment (diegoabt) [reply probinet/input/loader.py:11]: I will take care of this!

File: tests/test_input.py (hunk 11 above)
Comment (diegoabt) [reply tests/test_input.py:9]: I will get this once it's merged, as we discussed.

File: tests/test_input.py (hunk 12 above)
Comment (kibidi) [reply tests/test_input.py:23]: Great, thanks

**Comment by kibidi** [comment]
@jcpassy, thank you very much! It was fun!

**Pull request merged**